### PR TITLE
Fix #57232 potential CancellationTokenSource leak in HubConnection

### LINKED ISSUE #57232: CancellationTokenSource potential leaks in SignalR HubConnection
### Describe the bug

there is `CancellationTokenSource` type variable in src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

https://github.com/dotnet/aspnetcore/blob/257d69079e0f7fc84e3f6cd5047272d7f79b4d66/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs#L1599-L1600

and in `finally` block of this private method only Cancel() method calls

https://github.com/dotnet/aspnetcore/blob/257d69079e0f7fc84e3f6cd5047272d7f79b4d66/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs#L1704

How about to Dispose this CTS object in finally block or add `using` construction in its declaration to eliminate potential memory leaks?

Found by Linux Verification Center (linuxtesting.org) with SVACE.

### PR DESCRIPTION
**Pull request by AshkanAfsharpour**
# Fix potential CancellationTokenSource leak in HubConnection.cs #57232

- [x] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [x] You've included unit or integration tests for your change, where applicable.
- [x] You've included inline docs for your change, where applicable.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

## Summary of the changes (Less than 80 chars)

Fixes potential memory leak by ensuring proper disposal of `CancellationTokenSource` in `HubConnection.cs`.

## Description

This PR addresses issue #57232 regarding a potential memory leak in `HubConnection.cs` where a `CancellationTokenSource` is created but only `Cancel()` is called without disposing it. As per the discussion, I've refactored the code to use the `using var` pattern to ensure automatic disposal of the `CancellationTokenSource`.

Fixes #57232.

### CODE REVIEW

**Comment by AshkanAfsharpour** [comment]
> @AshkanAfsharpour please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @dotnet-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @dotnet-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @dotnet-policy-service agree company="Microsoft"
> > ```



> @dotnet-policy-service agree

@dotnet-policy-service agree

File: src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs [hunk 1]
```diff
@@ -1596,113 +1596,115 @@ private async Task ReceiveLoop(ConnectionState connectionState)
         var timer = new TimerAwaitable(TickRate, TickRate);
         var timerTask = connectionState.TimerLoop(timer);
 
-        var uploadStreamSource = new CancellationTokenSource();
-        connectionState.UploadStreamToken = uploadStreamSource.Token;
-        var invocationMessageChannel = Channel.CreateUnbounded<InvocationMessage>(_receiveLoopOptions);
+        using (var uploadStreamSource = new CancellationTokenSource())
```
Comment (BrennanConroy) [inline src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs:1599]: Use the `using var name = new type();` syntax. You don't need the parenthesis and additional formatting.
https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0063

Comment (AshkanAfsharpour) [reply src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: Thank you for the feedback, Brennan! I've updated the code to use the using var syntax as suggested. Please let me know if there's anything else that needs adjustment.

File: src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs [hunk 2]
```diff
@@ -1704,6 +1704,7 @@ async Task StartProcessingInvocationMessages(ChannelReader<InvocationMessage> in
             uploadStreamSource.Cancel();
             await HandleConnectionClose(connectionState).ConfigureAwait(false);
         }
+
```
Comment (BrennanConroy) [inline src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs:1707]:
```suggestion
```

**Review by BrennanConroy — APPROVED** [review]

**Comment by BrennanConroy** [comment]
Thanks @AshkanAfsharpour

**Pull request merged**